### PR TITLE
feat: add persistent logging for memberships

### PR DESCRIPTION
**Pull request by leogermani**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Adds log. See context on  peFeUf-i7-p2

### How to test the changes in this Pull Request:

Testing instruction peFeUf-i7-p2

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: includes/plugins/class-woocommerce-memberships.php
Comment (adekbadek) [inline includes/plugins/class-woocommerce-memberships.php:148]: Typo: here and in three other instances.

```suggestion
		Newspack_Newsletters_Logger::remote_log( 'newsletters_membership_handling', 'Membership status changed', $user_email, [ 'data' => $log_data ] );
```

**Review by adekbadek — APPROVED** [review]
Nit: I'm not convinced that implementation details of Newspack Manager logging should be exposed here. I'd rather see this implemented with action hooks.

Pre-approving pending the typo fix.

**Comment by leogermani** [comment]
> Nit: I'm not convinced that implementation details of Newspack Manager logging should be exposed here. I'd rather see this implemented with action hooks.

This is not a Nit, It's a valid point.

I'm gonna put this on hold so we can discuss. And I think we can revisit the need for it given the new finds on the investigation?

**Comment by leogermani** [comment]
Closing in favor of https://github.com/Automattic/newspack-newsletters/pull/1574

**Pull request closed without merging**
